### PR TITLE
Fix datadog.yaml template 

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

This PRs fixes a series of issue in our tempalte: missing values, misspelled settings, wrong indentation, missmatch settings name with the actual code, ...

### Describe how you validated your changes

No QA needed, this only touche the tempalte

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1197,7 +1197,7 @@ api_key:
   #         remove_query_string: false
   ##        @param DD_APM_OBFUSCATION_HTTP_REMOVE_PATHS_WITH_DIGITS - boolean - optional
   ##        If enabled, path segments in URLs containing digits are replaced by "?"
-  #         remove_path_with_digits: false
+  #         remove_paths_with_digits: false
```
Comment (hush-hush) [inline pkg/config/config_template.yaml:1200]: Code declaring this is [here](https://github.com/DataDog/datadog-agent/blob/ba08dfcc0bb4b569fc5a59e65652b46eeeba42a4/pkg/config/setup/apm.go#L40)

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -3613,7 +3619,7 @@ api_key:
 ## @param prometheus_scrape - custom object - optional
 ## This section configures the Autodiscovery based on the Prometheus annotations
 #
-# prometheus:
+# prometheus_scrape:
```
Comment (hush-hush) [inline pkg/config/config_template.yaml:3622]: This section was named differently in the code: https://github.com/DataDog/datadog-agent/blob/ba08dfcc0bb4b569fc5a59e65652b46eeeba42a4/pkg/config/setup/config.go#L395-L398

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -3864,100 +3870,100 @@ api_key:
       ## @param ignored_ip_addresses - list of strings - optional
       ## A list of IP addresses to ignore when scanning the network.
       #
-      # ignored_ip_addresses:
-      #   - <IP_ADDRESS_1>
-      #   - <IP_ADDRESS_2>
+      #   ignored_ip_addresses:
+      #     - <IP_ADDRESS_1>
+      #     - <IP_ADDRESS_2>
```
Comment (hush-hush) [inline pkg/config/config_template.yaml:3875]: This section and the following are part of a list started line 3868, so the indent need to reflect that.

**Review by michaelcretzman — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Review by stanistan — APPROVED** [review]

**Pull request merged**